### PR TITLE
Bump to Airflow 1.10.5

### LINKED ISSUE #435: Docker for Airflow 1.10.5
Not a major, but can we have an updated version? This release fixes several bugs in a few operators...

### PR DESCRIPTION
**Pull request by maxbrunet**
* Update Airflow version in Docker image
* Update Airflow image in `docker-compose` files
* Update `airflow.cfg` with changes from 1.10.5 (fixes a few not-templated values)
* Update Airflow version test in Circle CI

Closes #435

### CODE REVIEW

**Comment by Lord-Y** [comment]
@puckel Any plan to merge soon?

File: config/airflow.cfg [hunk 1]
```diff
@@ -30,27 +30,30 @@ logging_config_class =
 # Log format
 # Colour the logs when the controlling terminal is a TTY.
 colored_console_log = True
-colored_log_format = [%%(blue)s%%(asctime)s%%(reset)s] {{%%(blue)s%%(filename)s:%%(reset)s%%(lineno)d}} %%(log_color)s%%(levelname)s%%(reset)s - %%(log_color)s%%(message)s%%(reset)s
+colored_log_format = [%%(blue)s%%(asctime)s%%(reset)s] {%%(blue)s%%(filename)s:%%(reset)s%%(lineno)d} %%(log_color)s%%(levelname)s%%(reset)s - %%(log_color)s%%(message)s%%(reset)s
```
Comment (prideloki) [inline config/airflow.cfg]: could you point out to the changelog that mentions this update?
i looked up on https://github.com/apache/airflow/blob/1.10.5/airflow/config_templates/default_airflow.cfg#L64

there is no change on there

**Comment by apurvis** [comment]
> Any plan to merge soon?

+1 to this question

**Comment by mikaandria** [comment]
+1 for the merge

**Comment by genyrosk** [comment]
@puckel any plans on merging soon ?

File: config/airflow.cfg [hunk 2]
```diff
@@ -30,27 +30,30 @@ logging_config_class =
...
 log_filename_template = {{ ti.dag_id }}/{{ ti.task_id }}/{{ ts }}/{{ try_number }}.log
 log_processor_filename_template = {{ filename }}.log
 dag_processor_manager_log_location = /usr/local/airflow/logs/dag_processor_manager/dag_processor_manager.log
 
 # Hostname by providing a path to a callable, which will resolve the hostname
+# The format is "package:function". For example,
+# default value "socket:getfqdn" means that result from getfqdn() of "socket" package will be used as hostname
+# No argument should be required in the function specified.
+# If using IP address as hostname is preferred, use value "airflow.utils.net:get_host_ip_address"
 hostname_callable = socket:getfqdn
```
Comment (zenweasel) [inline config/airflow.cfg]: This is an extremely helpful note here

File: config/airflow.cfg (hunk 1 above)
Comment (KostyaEsmukov) [reply config/airflow.cfg]: This change is correct. In fact, what is currently in master seems to be wrong.

The config you're referencing is a jinja2 template, which after being rendered strips out some braces. Here's how you can easily retrieve a rendered version of the file you've mentioned:

```
$ docker run -it --rm -v `pwd`/config:/conf python:3.7 bash
root@91dd867c925b:/# pip install apache-airflow==1.10.5
root@91dd867c925b:/# airflow initdb
root@91dd867c925b:/# cp ~/airflow/airflow.cfg /conf/airflow_original.cfg
```

And here's the diff between that `airflow_original.cfg` and the file in this branch:

```patch
5c5
< dags_folder = /usr/local/airflow/dags
---
> dags_folder = /root/airflow/dags
9c9
< base_log_folder = /usr/local/airflow/logs
---
> base_log_folder = /root/airflow/logs
42c42
< dag_processor_manager_log_location = /usr/local/airflow/logs/dag_processor_manager/dag_processor_manager.log
---
> dag_processor_manager_log_location = /root/airflow/logs/dag_processor_manager/dag_processor_manager.log
62c62
< # sql_alchemy_conn = sqlite:////tmp/airflow.db
---
> sql_alchemy_conn = sqlite:////root/airflow/airflow.db
118c118
< plugins_folder = /usr/local/airflow/plugins
---
> plugins_folder = /root/airflow/plugins
121c121
< fernet_key = $FERNET_KEY
---
> fernet_key = 
258c258
< expose_config = True
---
> expose_config = False
372c372
< broker_url = redis://redis:6379/1
---
> broker_url = sqla+mysql://airflow:airflow@localhost:3306/airflow
380c380
< result_backend = db+postgresql://airflow:airflow@postgres/airflow
---
> result_backend = db+mysql://airflow:airflow@localhost:3306/airflow
478c478
< child_process_log_directory = /usr/local/airflow/logs/scheduler
---
> child_process_log_directory = /root/airflow/logs/scheduler
681c681
< # i.e. /usr/local/airflow/dags
---
> # i.e. /root/airflow/dags
829c829
< # Should be supplied in the format: key = value
---
> # Should be supplied in the format: key = value
\ No newline at end of file
```

**Comment by DannyNemer** [comment]
Hi, @puckel! What can I do to help you merge this PR? Thank you!

**Comment by ab-w** [comment]
Guys, there has been no activity from puckel whatsoever on GitHub in the past few months. Yes, there was that one commit in August, but that's it. This repo is already two Airflow versions and one Python version behind. There are dozens of open PRs and over a hundred open issues. It's time to move on! If you keep your hopes up, you'll only be disappointed. Maybe form a GitHub organization with a persistent 3 active admins, fork this repo there, and require each PR to be approved by ≥3 senior reviewers. Currently this repo is a bottleneck.

**Comment by datarods-svc** [comment]
Because puckel has been chronically inactive for months, and given that there are dozens of open PRs for this repo, I have created a new GitHub and Docker organization called `datarods` in which to collaboratively maintain a fork of this project further. There will be multiple owners of these organizations, thereby preventing a single inactive user from holding the repo hostage.

@maxbrunet Please feel encouraged to submit this PR there!

Please propose two new admins for the GitHub and DockerHub organizations!

**Comment by datarods-svc** [comment]
In the fork, I also now have a working [CI pipeline using GitHub Actions](https://github.com/datarods/docker-airflow/actions). I also made a new tagged release there with a corresponding automated [image on Docker Hub](https://hub.docker.com/r/datarods/docker-airflow/tags). **Please join me there with your much-needed contributions and reviews!**

**Comment by KostyaEsmukov** [comment]
For anyone considering switching to other docker images I would suggest to follow the upstream efforts to create an official production-ready Docker image:
- https://lists.apache.org/thread.html/fc2e1a579c8b21d464f4b8fef4647e3649e5b338052847a397ed0715@%3Cdev.airflow.apache.org%3E
- https://github.com/apache/airflow/pull/6266

Airflow devs already build some docker images for testing purposes at https://hub.docker.com/r/apache/airflow but AFAICT they're not production-ready.

I believe that instead of creating and maintaining more opensourced forks it would be much more productive to offer some help to the Airflow devs towards creating a single official production-ready Docker image.

@datarods-svc I had to deny your unsolicited invite to join that organization because it is not in my best interests.

**Comment by kabirbaidhya** [comment]
Any recent updates? Anything stopping this from getting merged?

**Comment by DannyNemer** [comment]
@puckel, why?!?!

**Comment by DannyNemer** [comment]
Ah ha - I see `master`. Thank you, @puckel!

File: config/airflow.cfg (hunk 2 above)
Comment (impredicative) [reply config/airflow.cfg]: Fwiw, this note is not included in https://github.com/puckel/docker-airflow/tree/1.10.6-1

**Pull request closed without merging**